File: functions/functions2/ex06.c
/*Write a function to check whether a number is in a given range*/

#include <stdio.h>

/* Say if a number is a specific range */
void in_range(int n1, int n2, int number);

int main()
{
    int start,end,number;

    printf("Insert the start number of the range (int): ");
    scanf("%i",&start);

    printf("Insert the last number of the range (int): ");
    scanf("%i",&end);

    printf("Insert a number (int): ");
    scanf("%i",&number);

    in_range(start,end,number);

    return 0;
}

void in_range(int n1, int n2, int number)
{   
    int key = 0;

    if (n1 < n2)
    {
        for (int i = n1; i < n2; i++)
        {
            if (i == number)
            {
                printf("%i is in the range\n",number);
                key++;
                break;
            }
        }
    }
    else
    {
        for (int i = n2; i < n1; i++)
        {
            if (i == number)
            {
                printf("%i is in the range\n",number);
                key++;
                break;
            }
        }
    }

    if (key == 0)
    {
        printf("%i is not in the range\n",number);
    }
}

File: functions/functions2/ex06.py
"""Write a function to check wheter a number is in a given range"""

def in_range(n1,n2,element):
    if n1 > n2:
        if element in [i for i in range(n2,n1+1)]:
            print(element,"is in range")
        else:
            print(element,"is not in range")
    if n2 > n1:
        if element in [i for i in range(n1,n2+1)]:
            print(element,"is in range")
        else:
            print(element,"is not in range")
#---------------------------------------------------------
num1 = int(input("Insert a number for range: "))
num2 = int(input("Insert other number for delimit the range: "))

num3 = int(input("Insert a number: "))

in_range(num1,num2,num3)
